site/src/component/SearchModule/SearchModule.tsx

```diff
@@ -22,6 +23,7 @@ const SearchModule: FC<SearchModuleProps> = ({ index }) => {
     const dispatch = useAppDispatch();
     const courseSearch = useAppSelector(state => state.search.courses);
     const professorSearch = useAppSelector(state => state.search.professors);
+    const [ pendingRequest, setPendingRequest ] = useState<NodeJS.Timeout | null>(null);
```

Comment:
We are not rendering anything based on the pending request, so this variable does not have to be stateful. We can just instantiate it as a normal variable.